cocos2d/effects/CCGrid.js

```diff
@@ -193,9 +213,16 @@ cc.GridBase = cc.Class.extend(/** @lends cc.GridBase# */{
         this._gridSize = gridSize;
         this._texture = texture;
         this._isTextureFlipped = flipped;
+        if(rect == undefined || cc._rectEqualToZero(rect))
```

Comment:
We don't do much defensive code in JS, so we only treat the case where rect is `undefined`. If it equals `null`, that means user passed null into this function, then it's his responsibility to correct the code.

The reason we use `===` instead of `==` is that `===` is more efficient and make no side affect